### PR TITLE
Spinner size followup (using design tokens)

### PR DESCRIPTION
**Pull request by msmithNI**
# Pull Request

## 🤨 Rationale

Associated feature 
- #346 

Nathan Mitchell, Fred Visser, Jesse Attas, and myself had a followup discussion about spinner sizes. The end result was that we wanted it to be easier for users to use the designer-prescribed spinner sizes. Nathan followed up with Brandon O'Keefe to confirm the 3 spinner sizes desired (they still went with 16/32/64, which also matches the spec).  
Separately I looked at the spinner usages in SystemLink Enterprise, and the vast majority are also 16x16, so I'm keeping 16x16 the default size as it was previously.

After discussion with the dev team, we decided to use design tokens to provide the designer-provided `small` `medium` and `large` sizes (instead of a `size` DOM attribute). Consumers can set the `height` to a token as outlined in the spinner Storybook docs/examples.

## 👩‍💻 Implementation

- Add design tokens `spinnerSmallHeight` (16px), `spinnerMediumHeight` (32px), and `spinnerLargeHeight` (64px). 16px (small) is the default, so using that token is optional.
- Add `aspect-ratio: 1 / 1` to the spinner styling, so consumers can just set height (instead of both width and height)
- Update Storybook docs to cover the 3 sizes and explain how to set them, and some examples of where they should be used in a page
- Update theme matrix story to have those 3 sizes

## 🧪 Testing

- Autotests
- Looked at updated stories

## ✅ Checklist

- [x] I have updated the project documentation to reflect my changes or determined no changes are needed.

### CODE REVIEW

File: packages/nimble-components/src/spinner/styles.ts [hunk 1]
```diff
@@ -12,8 +12,8 @@ export const styles = css`
     ${display('inline-flex')}
 
     :host {
-        width: 16px;
         height: 16px;
```
Comment (jattasNI) [inline packages/nimble-components/src/spinner/styles.ts:15]: I have a mild preference to create a "small" design token and use it to set the height here, assuming that doesn't break any other behaviors.

This preference is just based on a desire for completeness and reducing the hardcoded magic constants in our CSS.

File: packages/nimble-components/src/spinner/tests/spinner.stories.ts [hunk 2]
```diff
@@ -3,12 +3,39 @@ import type { Meta, StoryObj } from '@storybook/html';
...
-const metadata: Meta = {
+interface SpinnerArgs {
+    size: keyof typeof spinnerSize;
+}
+
+const overviewText = '<p>The `nimble-spinner` is an animating indicator that can be placed in a particular region of a page to represent '
+    + 'loading progress, or an ongoing operation, of an indeterminate / unknown duration.</p>'
+    + '<p>To customize its size, set its CSS `height` to a design token, and its width will automatically match its height.</p>'
+    + '<ul>'
+    + '<li>16x16 (small, default)'
```
Comment (jattasNI) [inline packages/nimble-components/src/spinner/tests/spinner.stories.ts:25]: Right now we're inconsistent about whether the primary API is "small/medium/large" or "16/32/64". In this file and the matrix story the pixel sizes are prominent and the t-shirt sizes are extra information, but in the design token documentation it's the opposite. I'm glad we describe both in most places, just think we should be consistent about which one is our "official" terminology.

I'd vote for small/medium/large to be the "official" naming, which would mean changing a few places in this file and the matrix story to be more like "medium (32x32)".

File: packages/nimble-components/src/spinner/tests/spinner.stories.ts [hunk 3]
```diff
@@ -3,12 +3,39 @@ import type { Meta, StoryObj } from '@storybook/html';
...
+} as const;
 
-const metadata: Meta = {
+interface SpinnerArgs {
+    size: keyof typeof spinnerSize;
+}
+
+const overviewText = '<p>The `nimble-spinner` is an animating indicator that can be placed in a particular region of a page to represent '
+    + 'loading progress, or an ongoing operation, of an indeterminate / unknown duration.</p>'
+    + '<p>To customize its size, set its CSS `height` to a design token, and its width will automatically match its height.</p>'
```
Comment (jattasNI) [inline packages/nimble-components/src/spinner/tests/spinner.stories.ts:23]: These docs (and the use of aspect ratio) are great. Should we do the same for icons? If this requires more discussion we could split the idea into a separate PR/issue.

**Review by jattasNI — APPROVED** [review]

File: packages/nimble-components/src/spinner/tests/spinner.stories.ts (hunk 3 above)
Comment (rajsite) [inline packages/nimble-components/src/spinner/tests/spinner.stories.ts:23]: I think all the specific height detail is too much discussion / duplicated detail for the overview text. In the overview text I think it should just say see the `size` details for customizing the spinner size.

**Review by rajsite — APPROVED** [review]

File: packages/nimble-components/src/spinner/tests/spinner.stories.ts [hunk 4]
```diff
@@ -22,15 +49,36 @@ const metadata: Meta = {
...
+                + `<li>For 32x32: \`height: var(${spinnerMediumHeight.cssCustomProperty});\`</li>`
+                + `<li>For 64x64: \`height: var(${spinnerLargeHeight.cssCustomProperty});\`</li></ul>`,
+            options: Object.keys(spinnerSize),
+            table: { defaultValue: { summary: '16x16 (small)' } },
+            control: {
+                type: 'radio',
+                labels: {
+                    small: '16x16 (small, default)',
+                    medium: `32x32 (${spinnerMediumHeight.cssCustomProperty})`,
+                    large: `64x64 (${spinnerLargeHeight.cssCustomProperty})`
```
[the remark this answers is not in the document]
Comment (msmithNI) [reply packages/nimble-components/src/spinner/tests/spinner.stories.ts]: Updated

File: packages/nimble-components/src/spinner/tests/spinner.stories.ts [hunk 5]
```diff
@@ -22,15 +49,36 @@ const metadata: Meta = {
                 'https://xd.adobe.com/view/33ffad4a-eb2c-4241-b8c5-ebfff1faf6f6-66ac/screen/dece308f-79e7-48ec-ab41-011f3376b49b/specs/'
         }
     },
-    argTypes: {},
+    argTypes: {
+        size: {
+            description:
+                'Size of the spinner. To customize its size, set its CSS `height` to a design token, and its width will automatically match its height:<br/><ul>'
+                + `<li>For 32x32: \`height: var(${spinnerMediumHeight.cssCustomProperty});\`</li>`
+                + `<li>For 64x64: \`height: var(${spinnerLargeHeight.cssCustomProperty});\`</li></ul>`,
```
[the remark this answers is not in the document]
Comment (msmithNI) [reply packages/nimble-components/src/spinner/tests/spinner.stories.ts]: Updated

File: packages/nimble-components/src/spinner/tests/spinner.stories.ts (hunk 3 above)
Comment (msmithNI) [reply packages/nimble-components/src/spinner/tests/spinner.stories.ts]: Updated

File: packages/nimble-components/src/spinner/tests/spinner.stories.ts (hunk 2 above)
Comment (msmithNI) [reply packages/nimble-components/src/spinner/tests/spinner.stories.ts]: Updated

File: packages/nimble-components/src/spinner/tests/spinner.stories.ts (hunk 3 above)
Comment (msmithNI) [reply packages/nimble-components/src/spinner/tests/spinner.stories.ts]: #992
